app/helpers/user_blocks_helper.rb

```diff
@@ -20,7 +20,7 @@ def block_status(block)
       # the max of the last update time or the ends_at time is when this block finished
       # either because the user viewed the block (updated_at) or it expired or was
       # revoked (ends_at)
-      last_time = [block.ends_at, block.updated_at].max
+      last_time = block.deactivates_at || [block.ends_at, block.updated_at].max
```

Comment:
Do we need the second half of this? It's just for old blocks because we're not backfilling `deactivates_at` for them?

Reply:
Yes, this is until backfilling is done. After that the logic will be simpler everywhere. `last_time` is just `block.deactivates_at`, `block.active?` is comparing the current time with `block.deactivates_at`.